great_expectations/rule_based_profiler/rule/rule.py

```diff
@@ -112,8 +112,8 @@ def run(
         """
         variables = build_parameter_container_for_variables(
             variables_configs=reconcile_rule_variables(
-                variables=variables,
-                variables_config=convert_variables_to_dict(variables=self.variables),
+                variables=self.variables,
+                variables_config=convert_variables_to_dict(variables=variables),
```

Comment:
This bug resulted in auto profiler variables not being substituted at runtime.

Reply:
@NathanFarmer I believe that this may not necessarily be the correct fix.  The logic for variables reconciliation was put together for a reason.  We actually hold the "overrides", supplied in the method arguments, as the "base". 
 Could we please consider solving the issue of the default being overwritten differently?

Reply:
@alexsherstinsky can you help me understand how using the "overrides" and then updating them (reconciliation strategy is "update") with the defaults would work? It seems to produce the incorrect profiler configuration by always using the defaults that have been specified. It seems to me that the test previously passed due to chance, but once I updated the default config the unintended behavior was revealed.

Reply:
@alexsherstinsky Also, this bug seems to be completely separate and apart from the state issues we have seen with using a custom config before using the default config or a different custom config.